yt_dlp/extractor/theplatform.py

```diff
@@ -297,6 +300,27 @@ def _real_extract(self, url):
 
         formats, subtitles = self._extract_theplatform_smil(smil_url, video_id)
 
+        note = 'Downloading m3u8 information'
+        errnote = 'Failed to download m3u8 information'
+        # on at least one site reading just the SMIL data misses some formats as well as subtitles.
+        # Therefore also read the m3u8 file and add those formats and subtitles
+        m3u8_url = url.split("&")[0]
+        res = self._download_webpage_handle(
```

Comment:
Another reason we should do this only conditionally: ThePlatform can serve direct mp4/mp3/etc links. Calling `_download_webpage_handle` on a mangled URL could result in the extractor downloading a 100MB+ mp4 file.

Reply:
OK, makes sense. I should be able to find a way of resolving the URL without actually downloading it... I'll check it in the next couple of days.

Reply:
an example of what I mean
```
ext = urlhandle_detect_ext(self._request_webpage(HEADRequest(url), video_id, fatal=False))
if ext == 'm3u8':
    # ...
```

Reply:
OK, that simplifies it for me, thanks!